starters/graphql-serverless-contentful/src/handlers/sqs-generate-job.test.ts

```diff
@@ -0,0 +1,83 @@
+import {
+	APIGatewayProxyEvent,
+	Context,
+	Callback,
+	APIGatewayProxyResult,
+} from 'aws-lambda';
+import { sendMessage } from '../utils/sqs/sendMessage';
+import { handler } from './sqs-generate-job';
+
+// jest.mock('@serverless-template/aws');
+
+describe('generateDemoJob', () => {
+	let subject: void | APIGatewayProxyResult;
+	let mathRandomMock: jest.SpyInstance<number, []>;
+
+	const sendMessageMock = jest.mocked(sendMessage, { shallow: true });
+
+	beforeAll(() => {
+		mathRandomMock = jest.spyOn(global.Math, 'random').mockReturnValue(0.11);
+	});
+
+	afterAll(() => {
+		mathRandomMock.mockRestore();
+		jest.clearAllMocks();
+	});
+
+	describe('when the message is sent sucessfully', () => {
+		beforeAll(async () => {
+			sendMessageMock.mockResolvedValue({
+				success: true,
+				data: {
+					MessageId: '123456789',
+				},
+			});
+			subject = await handler(
+				{} as APIGatewayProxyEvent,
+				{} as Context,
+				{} as Callback
+			);
+		});
+
+		afterAll(() => {
+			sendMessageMock.mockClear();
+		});
+
+		it('returns a 200 status code', () => {
+			if (subject) expect(subject.statusCode).toBe(200);
```

Comment:
`subject` should be set here. If it's not set, `handler` is not behaving as expected.

Consider the scenario where something goes wrong and `subject` is not set. The if-statement will skip the assertion, and Jest will consider the case handled. It might give a warning for the empty test, but IIRC, it will not fail.

We can either drop the if-statement. Or, if this is a TypeScript thing, we can use `assert` to ensure `subject` is set and tell Jest when it's not.

```javascript
import assert from 'assert';

it('now uses assert! 🎉', () => {
  assert(subject != null); // type unsafe check to match all null-y values
  // TypeScript is now happy
})
```